src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/Everything/PatientEverythingService.cs

```diff
@@ -0,0 +1,369 @@
+                        : await SearchCompartmentWithoutDate(resourceType, resourceId, since, type, encodedInternalContinuationToken, cancellationToken);
+
+                    if (!searchResult.Results.Any())
+                    {
+                        token.Phase = 3;
+                        goto case 3;
+                    }
+
+                    break;
+                case 3:
+                    searchResult = await SearchRevinclude(resourceId, since, type, encodedInternalContinuationToken, cancellationToken);
+                    break;
+                default:
+                    return new SearchResult(Enumerable.Empty<SearchResultEntry>(), null, null, Array.Empty<Tuple<string, string>>());
+            }
+
+            string newContinuationToken = null;
+            if (searchResult.ContinuationToken != null)
+            {
+                newContinuationToken = EverythingOperationContinuationToken.ToString(token.Phase, searchResult.ContinuationToken);
+            }
+            else if (token.Phase < 3)
+            {
+                newContinuationToken = EverythingOperationContinuationToken.ToString(token.Phase + 1, null);
+            }
+
+            return new SearchResult(searchResult.Results, newContinuationToken, searchResult.SortOrder, searchResult.UnsupportedSearchParameters);
+        }
+
+        private async Task<SearchResult> SearchIncludes(
+            string resourceType,
+            string resourceId,
+            PartialDateTime since,
+            string type,
+            CancellationToken cancellationToken)
+        {
+            using IScoped<ISearchService> search = _searchServiceFactory();
+            var searchResultEntries = new List<SearchResultEntry>();
+
+            // Build search parameters
+            var searchParameters = new List<Tuple<string, string>>
+            {
+                Tuple.Create(SearchParameterNames.Id, resourceId),
+            };
+
+            searchParameters.AddRange(_includes.Select(include => Tuple.Create(SearchParameterNames.Include, $"{resourceType}:{include}")));
+
+            // Search
+            SearchOptions searchOptions = _searchOptionsFactory.Create(resourceType, searchParameters);
+            SearchResult searchResult = await search.Value.SearchAsync(searchOptions, cancellationToken);
+            searchResultEntries.AddRange(searchResult.Results.Select(x => new SearchResultEntry(x.Resource)));
+
+            // Filter results by IsDeleted
+            searchResultEntries = searchResultEntries.Where(s => !s.Resource.IsDeleted).ToList();
+
+            // Filter results by _type
+            if (!string.IsNullOrEmpty(type))
+            {
+                IReadOnlyList<string> types = type.SplitByOrSeparator();
+                searchResultEntries = searchResultEntries.Where(s => types.Contains(s.Resource.ResourceTypeName)).ToList();
+            }
+
+            // Filter results by _since
+            if (since != null)
+            {
```

Comment:
You had code in previous iteration which constructed expressions, you can just reuse same code, and create manually SearchOptions object rather than call searchOptionFactory.

That way you filter out results on server rather than application level.

Reply:
I'm not sure whether I get the point correctly. Here we do some post-process to filter out the results because resources pulled by `_include` are "less controlled". For example, it is not limited by `_since`. So we just remove those resources in application level.